pkg/cloudprovider/aws/packing/nodecapacity.go

```diff
@@ -15,77 +15,37 @@ limitations under the License.
 package packing
 
 import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
 	resourcesUtil "github.com/awslabs/karpenter/pkg/utils/resources"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
-// TODO get this information from node-instance-selector
-var (
-	nodeCapacities = []*nodeCapacity{
+func instanceTypeInfoToNodeCapacity(instanceTypeInfo ec2.InstanceTypeInfo) *nodeCapacity {
+	instanceTypeName := *instanceTypeInfo.InstanceType
+	vcpusInMillicores := resource.MustParse(fmt.Sprint(*instanceTypeInfo.VCpuInfo.DefaultVCpus * 1000))
+	memory := resource.MustParse(fmt.Sprintf("%dMi", *instanceTypeInfo.MemoryInfo.SizeInMiB))
+	// The number of pods per node is calculated using the formula:
+	//   max number of ENIs * (IPv4 Addresses per ENI -1) + 2
+	// https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt#L20
+	podCapacity := *instanceTypeInfo.NetworkInfo.MaximumNetworkInterfaces*(*instanceTypeInfo.NetworkInfo.Ipv4AddressesPerInterface-1) + 2
+	podCapacityResource := resource.MustParse(fmt.Sprint(podCapacity))
```

Comment:
nit: I'd probably inline all/most of these (it's kind of the house style for K.)